README
ELF::Extract::Sections



INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for ELF-Extract-Sections

1.001005 2015-09-29T14:22:24Z 94326c6

1.001004 2015-09-29T14:21:04Z b0ca7c0
 [Tests]
 - Try to work out what the finger prints of a system that can't handle ELF64 files looks like.

1.001003 2015-09-27T08:06:04Z 55e4d01
 [Dependencies::Stats]
 - Dependencies changed since 1.001002, see misc/*.deps* for details
 - runtime: +1
 - test: -1

 [Win32 Support]
 - Handle reading from objdump using slurp style IO instead of using an IO Handle  due to open '-|', @list not being
   reliable on Win32.

1.001002 2015-09-27T07:39:23Z f807e17
 [Installation]
 - Checks now attempt to extract the version itself from the version string and versions prior to 2.17 are now all
   known bad.

1.001001 2015-09-26T06:22:10Z 5cf1ba0
 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - configure: +3 -1
 - develop: +3 ↑1 -1 (suggests: ↑1)

 [Installation]
 - Now have a cursory white/black list of known good/bad objdump versions.
 - Not yet possible to clearly make sense of how to parse objdump's version directly and its not yet known exactly
   which version introduces the required functions.

 [Installation: TODO]
 - work out how to extract the list of supported targets to make sure that the ELF target is supported, and that all
   the machine targets that the test files are coded in are supported.

1.001000 2015-08-14T06:32:18Z 4d78d05
 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - runtime: +1
 - test: +2

 [Internals]
 - Parameter handling delegated to MooseX::Params::Validate
 - TODO: guard against spurious bless() causing problems, re rt#106192

 [Tests]
 - Probing for version data to make a better verison check

1.000000 2015-07-30T16:49:16Z 1c37bc4
 [!Major]
 - This release includes some respectably significant changes.

 [01 Install Side Tooling change]
 - Previously, Module::Build was used as the installer.
 - This is now ExtUtils::MakeMaker
 - See https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [02 Version Scheme Change]
 - Previously used a lengthy 0.{'0' x '8'} scheme.
 - Now uses a 0.{'0' x 6} scheme for clarity.
 - The 1.x on the box is due in part to this change.
 - See: https://questhub.io/realm/perl/quest/5388d3a9bbd0be2224000129

 [03 Internals Overhaul]
 - MooseX::Declare is too pesky to use with perltidy (even sweetened)
 - So everything has been entirely rewritten in pure Moose with simple type checks.
 - Class::Load has been replaced with Module::Runtime

 [Dependencies::Stats]
 - Dependencies changed since 0.03000101, see misc/*.deps* for details
 - build: -1
 - configure: +3 -1 (recommends: +1)
 - develop: +55 -4 (recommends: +1 -1, suggests: +1 ↑1)
 - runtime: +3 -2 (recommends: +1)
 - test: +4 ↑1 ↓1 -2 (recommends: +4)

0.03000101 2013-05-07T08:56:28Z
 [!Major Changes]
 - Path::Class replaced with Path::Tiny
 - Tests/Utils use Path::Iterator::Rule instead of File::Find::Rule
 - Packaging toolchain updated
 - Carp warnings now only Log4Perl warnings so you can hide them if needed.

 [Dependencies::Stats]
 - Dependencies changed since 0.02071411, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +8 (recommends: ↑1, suggests: ↑1)
 - runtime: +3 -1
 - test: +2 ↑1 -3

 [Documentation]
 - license copyright year update
 - license reindent
 - license address update
 - add README.mkdn

 [Internals]
 - add $AUTHORITY to all packages
 - move $VERSION declaration out of BEGIN
 - __END__ before =pod
 - Code now tidyied successfully under Perl::Tidy::Sweetened

 [Meta]
 - add x_authortiy
 - bug tracker = github issues
 - git uris to https://

 [Packaging]
 - use test_requires in Build.PL
 - leave xt/ tests in xt/
 - versions via git

0.02071411 2011-04-04T08:40:21Z
 [BugFix]
 - Add temporary hack for rt#67210 - methods with return types fail in void context.
 - Fix some 'invalid' changelog entries.

 [Dependencies::Stats]
 - Dependencies changed since 0.02021113, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +1 (recommends: +1, suggests: +1)
 - runtime: +1 -7
 - test: +9

 [Internals]
 - Use Class::Load instead of eval for loading extensions.

 [Packaging]
 - Rework Changelog to CPAN::Changes form.
 - Move to @Author::KENTNL

 [Tests]
 - Remove hand written boilerplate/kwalitee/critic tests.
 - Ship new (generated) tests report-versions-tiny, cpan-changes, eol and kwalitee

0.02021113 2009-11-28T23:43:20Z
 - Hard Dep on Perl5.10 due to Advanced regex usage ( Thanks CPAN Testers )

 [Dependencies::Stats]
 - Dependencies changed since 0.02020505, see misc/*.deps* for details
 - build: +1
 - configure: +1
 - runtime: +1 ↑1

0.02020505 2009-11-22T02:28:46Z
 - Added license data for binary .so's used for testing. See LICENSE.tests, and LICENSES/

0.02020308 2009-11-20T05:05:28Z
 - Fixed t/ by using Compile Tests. ( You can thank JAWNSY for bugging me about this )
 - Moved To AutoVersion::Relative
 - Dropped EUMM

0.0105 2009-08-22T15:17:47Z
 - Tweaked Authortests/Release tests bundling somewhat so they actually work.
 - Added Perl Critic tests and satisfied criteria.
 - Documenting the hell out of the Objdump default  scanner, partially to see how CPAN renders this  abuse. With a bit
   of luck, it will produce mean  as API breakdowns for me :)

0.0104 2009-08-21T17:34:29Z
 - Fixed Extrac typo which lead to dual indexing.
 - Dist::Zilla automatic tests now included.

0.0103 2009-07-19T14:59:11Z
 - Muted warning in tests
 - Moved to Dist::Zilla instead of MI
 - Updated to MooseX::Has::Sugar 0.0300

0.0102 2009-05-16T04:19:31Z
 - Dependancy Fixups, yes, more of them.

0.0101 2009-05-16T03:46:44Z
 - Re-Release with fixed inc/
 - using MX::Has::Sugar
 - dropped Hard dependancy on Perl 5.10

0.01.00 2009-05-11T02:04:58Z
 - First Version. Mostly working.


Build.PL
use Devel::CheckBin;
check_bin('objdump');


# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.015.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4205"
  },
  "configure_requires" => {
    "Devel::CheckBin" => 0,
    "Module::Build" => "0.4205"
  },
  "dist_abstract" => "Extract Raw Chunks of data from identifiable ELF Sections",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>"
  ],
  "dist_name" => "ELF-Extract-Sections",
  "dist_version" => "0.03000102",
  "license" => "perl",
  "module_name" => "ELF::Extract::Sections",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Class::Load" => 0,
    "Log::Log4perl" => "1.21",
    "Moose" => "0.76",
    "MooseX::Declare" => "0.20",
    "MooseX::Has::Sugar" => "0.0300",
    "MooseX::Log::Log4perl" => "0.31",
    "MooseX::Types" => "0.10",
    "MooseX::Types::Moose" => "0.10",
    "MooseX::Types::Path::Tiny" => 0,
    "overload" => 0,
    "perl" => "5.010000",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "FindBin" => 0,
    "List::Util" => 0,
    "Log::Log4perl" => "1.21",
    "Path::Iterator::Rule" => 0,
    "Path::Tiny" => 0,
    "Test::More" => "1.001003",
    "YAML::XS" => 0,
    "lib" => 0,
    "version" => 0
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec::Functions" => 0,
  "FindBin" => 0,
  "List::Util" => 0,
  "Log::Log4perl" => "1.21",
  "Module::Build" => "0.4205",
  "Path::Iterator::Rule" => 0,
  "Path::Tiny" => 0,
  "Test::More" => "1.001003",
  "YAML::XS" => 0,
  "lib" => 0,
  "version" => 0
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
